bin/martinize2

```diff
@@ -83,19 +84,22 @@ def martinize(system, mappings, to_ff, delete_unknown=False):
     return system
 
 
-def write_gmx_topology(system, top_path):
+def write_gmx_topology(system, top_path, deduplicate=True):
     if not system.molecules:
         raise ValueError('No molecule in the system. Nothing to write.')
-    # Deduplicate the moleculetypes in order to write each molecule ITP only
-    # once.
-    molecule_types = [[system.molecules[0], [system.molecules[0]]], ]
-    for molecule in system.molecules[1:]:
-        for molecule_type, share_moltype in molecule_types:
-            if molecule.share_moltype_with(molecule_type):
-                share_moltype.append(molecule)
-                break
-        else:  # no break
-            molecule_types.append([molecule, [molecule, ]])
+    if deduplicate:
```

Comment:
This'll need to become a processor at some point. It'll be a painful one to write.